mathesar/api/serializers/constraints.py

```diff
@@ -1,13 +1,58 @@
 from rest_framework import serializers
 
+from mathesar.api.serializers.shared_serializers import (
+    MathesarPolymorphicErrorMixin,
+    ReadWritePolymorphicSerializerMappingMixin,
+)
 from mathesar.models import Constraint, Column
 
 
-class ConstraintSerializer(serializers.ModelSerializer):
+class BaseConstraintSerializer(serializers.ModelSerializer):
     name = serializers.CharField(required=False)
     type = serializers.CharField()
     columns = serializers.PrimaryKeyRelatedField(queryset=Column.current_objects.all(), many=True)
 
     class Meta:
         model = Constraint
         fields = ['id', 'name', 'type', 'columns']
+
+
+class ForeignKeyConstraintSerializer(BaseConstraintSerializer):
+    class Meta:
+        model = Constraint
+        fields = BaseConstraintSerializer.Meta.fields + [
+            'referent_columns',
+            'onupdate',
+            'ondelete',
+            'deferrable',
+            'match'
+        ]
+
+    referent_columns = serializers.PrimaryKeyRelatedField(queryset=Column.current_objects.all(), many=True)
+    onupdate = serializers.ChoiceField(choices=['CASCADE', 'DELETE', 'RESTRICT'])
+    ondelete = serializers.ChoiceField(choices=['CASCADE', 'DELETE', 'RESTRICT'])
```

Comment:
Assuming these are supposed to represent and cover the options for `ON UPDATE` and `ON DELETE` in the DB, I think these lists should be `['CASCADE', 'RESTRICT', 'NO ACTION', 'SET NULL', 'SET DEFAULT']`. Is `'DELETE'` supposed to be in place of `'SET NULL'`? See [the docs](https://www.postgresql.org/docs/13/ddl-constraints.html#DDL-CONSTRAINTS-FK).

Reply:
Thanks for pointing it out @mathemancer, I was using the options listed in[alembic docs](https://alembic.sqlalchemy.org/en/latest/ops.html?highlight=foreign#alembic.operations.Operations.create_foreign_key), with the assumption that they were using generic terms to make it compatible across databases, but it looks like it isn't the case.